pkg/daemon/daemon.go

```diff
@@ -803,6 +791,53 @@ func (dn *Daemon) storeCurrentConfigOnDisk(current *mcfgv1.MachineConfig) error
 	return writeFileAtomicallyWithDefaults(dn.currentConfigPath, mcJSON)
 }
 
+// ensureNodeStateAnnotations does nothing if the node already has a CurrentConfig
+// annotation.  Otherwise, it tries to read it from the file injected by the MCS,
+// or falls back to querying the journal.  The boolean return value will be
+// true if we detected that the MCS file exists; this signals a transition
+// point between the early bootstrap and the MCD managing the system.
+func (dn *Daemon) ensureNodeStateAnnotations() (bool, error) {
+	ccAnnotation, err := getNodeAnnotation(dn.node, constants.CurrentMachineConfigAnnotationKey)
+	// we need to load the annotations from the file only for the
+	// first run.
+	// the initial annotations do no need to be set if the node
+	// already has annotations.
+	if err == nil && ccAnnotation != "" {
+		return false, nil
+	}
+
+	glog.Infof("No %s annotation on node %s: %v, in cluster bootstrap, loading initial node annotation from %s", constants.CurrentMachineConfigAnnotationKey, dn.node.Name, dn.node.Annotations, constants.InitialNodeAnnotationsFilePath)
+
+	d, err := ioutil.ReadFile(constants.InitialNodeAnnotationsFilePath)
+	if err != nil && !os.IsNotExist(err) {
+		return false, fmt.Errorf("failed to read initial annotations from %q: %v", constants.InitialNodeAnnotationsFilePath, err)
+	}
```

Comment:
super silly nit: shouldn't we return `true` from now on here? I'm just worried in the future we're gonna change the outer logic and might miss the bootstrap bool

Reply:
Would that would apply for the `json.Unmarshal`  and `setNodeAnnotations(..., initial)` error cases as well?

Reply:
yes, from this point it should be safe to return true